poetry/repositories/repository.py

```diff
@@ -24,7 +24,7 @@ def __init__(self, packages: List["Package"] = None, name: str = None) -> None:
             self.add_package(package)
 
     @property
-    def name(self) -> str:
+    def name(self) -> Optional[str]:
```

Comment:
Likewise I'm not sure that this Optional is correct. I think the source stems somewhere else.

Reply:
ditto

Reply:
Above

Reply:
I think the proposed FIXME comment in pool.py should be sufficient: it'll be impossible to get rid of the Option there without considering this.

(For much the same reasons as elsewhere, I think it's valuable to have annotations that match the reality of the code, even if we do think that the code is suboptimal)